colossalai/kernel/cuda_native/csrc/cpu_adam.cpp

```diff
@@ -493,7 +493,7 @@ int adam_step(int optimizer_id,
                 grads_ptr,
                 exp_avg_ptr,
                 exp_avg_sq_ptr,
-                params_c.size(0),
+                params_c.numel(),
```

Comment:
@Gy-Lu How did your code pass unit tests if this is a bug?

Reply:
oh...
we don't use suah a huge data size in unit tests to make this parameter in use